### PR TITLE
added regex escaping for the member expressions

### PR DESCRIPTION
**Pull request by AlexMost**
Addressing this issue https://github.com/c-3po-org/babel-plugin-c-3po/issues/90
@alxpy

### CODE REVIEW

**Review by MrOrz — CHANGES_REQUESTED** [review]
I have tried the following test cases in `test_utils.js`:

```javascript
    it('should throw if not all expressions exist in translated strings', () => {
        const input = '${ fooxbar } apples (translated)';
        const func = () => validateAndFormatMsgid(input, ['foo.bar']);
        expect(func).to.throw(
            'NoExpressionError: Expression \'foo.bar\' is not found in the localized string ' +
            '\'${ fooxbar } apples (translated)\'.'
        );
    });

    it('should support [] in computed properties', () => {
        const input = "${ a[value[3]] } banana ${ b['[key]'] }";
        const expected = "`${ a[value[3]] } banana ${ b['[key]'] }`";
        expect(validateAndFormatMsgid(input, ['a[value[3]]', "b['[key]']"])).to.eql(expected);
    });

    it('should support string in computed properties', () => {
        const input = "${ a['^[a]$'] } banana";
        const expected = "`${ a['^[a]$'] } banana`";
        expect(validateAndFormatMsgid(input, ["a['^[a]$']"])).to.eql(expected);
    });

    it('should support template strings in computed properties', () => {
        const input = '${ a[`foo-${value}`] } banana';
        const expected = '`${ a[`foo-${value}`] } banana`';
        expect(validateAndFormatMsgid(input, ['a[`foo-${value}`]'])).to.eql(expected);
    });
```
All but the following test succeeded:
```
  1) utils validateAndFormatMsgid should support string in computed properties:

      AssertionError: expected '`${ a[\'^[a] banana] } banana`' to deeply equal '`${ a[\'^[a]$\'] } banana`'
      + expected - actual

      -`${ a['^[a] banana] } banana`
      +`${ a['^[a]$'] } banana`

      at Context.<anonymous> (tests/unit/test_utils.js:137:66)
```
I admit that the test case is crafted a little too extreme. Should we handle this?

File: src/utils.js [hunk 1]
```diff
@@ -72,7 +73,7 @@ const memoize1 = (f) => (arg) => {
 };
 
 const reg = (i) => new RegExp(`\\$\\{([\\s]+?|\\s?)${i}([\\s]+?|\\s?)}`);
```
Comment (MrOrz) [inline src/utils.js]: The rail graph of this regexp looks like (assume `i` is `"XXXX"`):
https://regexper.com/#%5C%24%5C%7B(%5B%5Cs%5D%2B%3F%7C%5Cs%3F)XXXX(%5B%5Cs%5D%2B%3F%7C%5Cs%3F)%7D

The capturing group is not used in where `memReg()` is invoked, and `\s+|\s?` and `\s*` should be equivalent. I've tried
```javascript
const reg = (i) => new RegExp(`\\$\\{\\s*${i}\\s*}`);
```
And it does not break any of the existing test cases.
Shall we simplify this regexp as well?

Comment (AlexMost) [reply src/utils.js]: seems it's reasonable to simplify that regexp.

**Comment by AlexMost** [comment]
```"${ a['^[a]$'] } banana";``` seems a little bit complicated example, but it will be good if we will also handle this. Will take a look soon. If you have some thoughts on how that should be handled - I will really appreciate :)

**Comment by AlexMost** [comment]
@MrOrz So here is my update:

1. All those regex replace logic was implemented to normalize spaces inside `${ }` expressions. (to prevent accident mistake of adding or removing space inside the curly braces by the translator. But `babel-generator` can already do that, so I decided to get rid of those regexes via:
```js
const msgidAST = tpl(strToQuasi(msgid))();
generate(msgidAST).code.replace(/;$/, '');
```
> babel-generator adds the semicolon at the end, we don't need that in our case.

2. Added all your test cases to the test_utils.js, so thanks a lot for them :)

Seems like finally, it should work )

**Pull request merged**